src/lib/Context.ts

```diff
@@ -119,14 +147,19 @@ export function consolidateConfigs<T>(...configs: Partial<ConfigJson>[]): Config
   }, {} as ConfigJson)
 }
 
-export const createContext = async (configPath: string, overrides: ConfigJson): Promise<Context> => {
+export const createContext = async (configPath: string, overrides: ConfigJson, transactionContext: boolean = false): Promise<Context> => {
   const config: ConfigJson = consolidateConfigs(DEFAULT_CONFIG, readConfig(configPath), overrides)
 
-  const signer = await createSignerFromPath(config.keypair!)
+  // if transactionContext is true, we need to make sure we have a valid keypair either in the config or overrides
+  if (transactionContext === true && !config.keypair && !overrides.keypair) {
```

Comment:
I suggest cleaning up the logic flow here, like:

```
if (transactionContext) {
  signer = await createSignerFromPath(config.keypair) // should throw good error message
} else  {
  signer = config.keypair ? await createSignerFromPath(config.keypair) : createNoopSigner()
}
```

or you can go even more consolidated:

```
signer = (isTransactionContext || !!config.keypair) ?  await createSignerFromPath(config.keypair) : createNoopSigner()
```